### PR TITLE
Change storage_v2 GetTrace to GetTraces plural

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- GetTrace returns a single trace. The use case for FindTraceIDs specifically exists for loading traces in batches, but there's really no capability in Jaeger to do that: if you loaded 10K IDs you need to make 10K RPCs to load 10K traces. 
- Part of #5079

## Description of the changes
- Change it to `GetTraces(...TraceID)`
- Also add start/end times (per #4150)

## How was this change tested?
- CI, unit tests

### CODE REVIEW

File: storage_v2/tracestore/reader.go [hunk 1]
```diff
@@ -35,32 +40,40 @@ type Reader interface {
...
 	// of matching trace IDs. This is useful in some contexts, such as batch jobs, where a
 	// large list of trace IDs may be queried first and then the full traces are loaded
 	// in batches.
-	FindTraceIDs(ctx context.Context, query TraceQueryParameters) iter.Seq2[[]pcommon.TraceID, error]
+	FindTraceIDs(ctx context.Context, query TraceQueryParams) iter.Seq2[[]pcommon.TraceID, error]
 }
 
-// TraceQueryParameters contains parameters of a trace query.
-type TraceQueryParameters struct {
+// GetTraceParams contains single-trace parameters for a GetTraces.
```
Comment (mahadzaryab1) [inline storage_v2/tracestore/reader.go:66]:
```suggestion
// GetTraceParams contains single-trace parameters for a GetTraces request.
```

File: storage_v2/tracestore/reader.go [hunk 2]
```diff
@@ -35,32 +40,40 @@ type Reader interface {
...
+// Some storage backends (e.g. Tempo) perform GetTraces much more efficiently
+// if they know the approximate time range of the trace.
+type GetTraceParams struct {
+	TraceID pcommon.TraceID // required
+	Start   time.Time       // optional
+	End     time.Time       // optional
```
Comment (mahadzaryab1) [inline storage_v2/tracestore/reader.go:72]: should we add proper godoc comments for these?

File: storage_v2/factoryadapter/reader_test.go [hunk 3]
```diff
@@ -62,9 +62,11 @@ func TestTraceReader_GetTraceDelegatesSuccessResponse(t *testing.T) {
 	traceReader := &TraceReader{
 		spanReader: sr,
 	}
-	traces, err := iter.CollectWithErrors(traceReader.GetTrace(
```
Comment (mahadzaryab1) [inline storage_v2/factoryadapter/reader_test.go:65]: is `CollectWithErrors` still being used?

File: storage_v2/factoryadapter/reader.go [hunk 4]
```diff
@@ -40,27 +40,36 @@ func NewTraceReader(spanReader spanstore.Reader) *TraceReader {
...
+func (tr *TraceReader) GetTraces(
+	ctx context.Context,
+	traceIDs ...tracestore.GetTraceParams,
+) iter.Seq2[[]ptrace.Traces, error] {
+	return func(yield func([]ptrace.Traces, error) bool) {
+		for _, idParams := range traceIDs {
+			// TODO start/end times are not supported by v1 reader
+			// https://github.com/jaegertracing/jaeger/pull/6242
+			t, err := tr.spanReader.GetTrace(ctx, model.TraceIDFromOTEL(idParams.TraceID))
+			if err != nil && !errors.Is(err, spanstore.ErrTraceNotFound) {
```
Comment (mahadzaryab1) [inline storage_v2/factoryadapter/reader.go:52]: do we have a test for the `ErrTraceNotFound` case?

Comment (mahadzaryab1) [reply storage_v2/factoryadapter/reader.go:52]: also - shouldn't we continue if the trace isn't found?

**Review by mahadzaryab1 — COMMENTED** [review]
few nits / questions

File: storage_v2/factoryadapter/reader_test.go (hunk 3 above)
Comment (yurishkuro) [reply storage_v2/factoryadapter/reader_test.go:65]: don't think so

Comment (yurishkuro) [reply storage_v2/factoryadapter/reader_test.go:65]: maybe not

File: storage_v2/factoryadapter/reader.go (hunk 4 above)
Comment (yurishkuro) [reply storage_v2/factoryadapter/reader.go]: good catch

File: storage_v2/factoryadapter/reader_test.go (hunk 3 above)
Comment (mahadzaryab1) [reply storage_v2/factoryadapter/reader_test.go:65]: can we remove it then?

**Review by mahadzaryab1 — APPROVED** [review]

File: storage_v2/factoryadapter/reader_test.go (hunk 3 above)
Comment (yurishkuro) [reply storage_v2/factoryadapter/reader_test.go:65]: It's not part of this PR, so I'd leave it. That whole package will go away in a couple of months.

**Pull request merged**